### PR TITLE
BUG: Fix new page form clearing when selecting a `Under another page` option

### PR DESCRIPTION
**Pull request by bergice**
Fixes #2339

Revert d05e1424ee1d7217474d7d16f8bdc4e289dd2700

### CODE REVIEW

**Comment by robbieaverill** [comment]
Does this mean we lose all the loading animations again? I’d rather have them than fix this bug, if I had to choose

**Comment by bergice** [comment]
> Does this mean we lose all the loading animations again? I’d rather have them than fix this bug, if I had to choose

Yep.

On closer look I can see this is caused by `2f2fb2bfdcd88b54ac9426f172bec77a3c210172` - it just replaces the content with the loading spinner rather than appending to the element.

Will update PR.

**Comment by robbieaverill** [comment]
Boom

File: client/src/legacy/LeftAndMain.js [hunk 1]
```diff
@@ -1048,17 +1048,16 @@ $.entwine('ss', function($) {
...
+        container[0]
       );
     },
     onunmatch: function() {
       this._super();
-      const container = this[0];
-      if (container) {
-        ReactDOM.unmountComponentAtNode(container);
-      }
+      this.find('.cms-loading-container').remove();
```
[the remark this answers is not in the document]
Comment (bergice) [reply client/src/legacy/LeftAndMain.js]: You're right. It needs this to be unmounted properly. I've updated the code.

**Comment by ScopeyNZ** [comment]
Ok I literally prepared this exact patch for another issue I've encountered where `gridfield-orderable-rows` is not working.

**Comment by ScopeyNZ** [comment]
The test failures are pre-existing in 1.3: https://travis-ci.org/silverstripe/silverstripe-admin/branches 

They're being looked at separately.

**Pull request merged**